x-pack/plugins/enterprise_search/server/lib/engines/field_capabilities.ts

```diff
@@ -33,46 +33,52 @@ export const fetchEngineFieldCapabilities = async (
   };
 };
 
-const ensureIndices = (indices: string | string[] | undefined): string[] => {
+const ensureIndices = (indices: string[] | string | undefined): string[] => {
   if (!indices) return [];
   return Array.isArray(indices) ? indices : [indices];
 };
 
-export const parseFieldsCapabilities = (
-  fieldCapsResponse: FieldCapsResponse,
-  prefix: string = ''
-): SchemaField[] => {
+export const parseFieldsCapabilities = (fieldCapsResponse: FieldCapsResponse): SchemaField[] => {
   const { fields, indices: indexOrIndices } = fieldCapsResponse;
-  const inThisPass: Array<[string, Record<string, FieldCapsFieldCapability>]> = Object.entries(
-    fields
-  )
-    .filter(([key]) => key.startsWith(prefix))
-    .map(([key, value]) => [key.replace(prefix, ''), value]);
+  const indices = ensureIndices(indexOrIndices);
 
-  const atThisLevel = inThisPass.filter(([key]) => !key.includes('.'));
+  return Object.entries(fields)
+    .map(([fieldName, typesObject]) => {
+      const typeValues = Object.values(typesObject);
+      const type = calculateType(Object.keys(typesObject));
 
-  return atThisLevel.map(([name, value]) => {
-    const type = calculateType(Object.keys(value));
-    let indices = Object.values(value).flatMap((fieldCaps) => {
-      return ensureIndices(fieldCaps.indices).map((index) => ({
-        name: index,
-        type: fieldCaps.type,
-      }));
-    });
+      const indicesToType = typeValues.reduce(
+        (acc: Record<string, string>, { type: indexType, indices: typeIndexOrIndices }) => {
+          const typeIndices = ensureIndices(typeIndexOrIndices);
+          typeIndices.forEach((index) => {
+            acc[index] = indexType;
+          });
+          return acc;
+        },
+        {}
+      );
 
-    indices =
-      indices.length === 0
-        ? ensureIndices(indexOrIndices).map((index) => ({ name: index, type }))
-        : indices;
+      const fieldIndices =
+        Object.keys(indicesToType).length > 0
+          ? indices.map((index) => {
+              const indexType = indicesToType[index] || 'unmapped';
```

Comment:
Nitpicky question, why don't we use the `UNMAPPED` constant defined in `engine_schema` here instead of the String?

Reply:
oh i was actually just using these to manually mark certain other types as conflicts or unmapped when building out UI. i'll just switch over to string literals